app/controllers/contents_controller.rb

```diff
@@ -2,20 +2,17 @@
 
 # Controller for a Work contents (files)
 class ContentsController < ApplicationController
-  before_action :set_content, only: %i[edit update show]
+  before_action :set_content, only: %i[edit update show show_table]
+  before_action :set_content_files, only: %i[show show_table]
 
+  # Called from work edit/update form.
   def show
     authorize! @content
+  end
 
-    @content_files = @content.content_files.order(filename: :asc).page(params[:page])
-
-    # This can be called from the work edit/update page or the work show page.
```

Comment:
When roundtripping failed, the user is redirected here. However, in this case the referrer test below fails and the wrong template was being rendered.